integration/querydsl/expressions/src/main/java/com/blazebit/persistence/querydsl/JPQLNextQuery.java

```diff
@@ -35,12 +36,11 @@
  * Query interface for JPQL.Next queries
  *
  * @param <T> Query result type
- * @param <Q> Concrete query type
  * @author Jan-Willem Gmelig Meyling
  * @since 1.5.0
  */
 @SuppressWarnings("unused")
-public interface JPQLNextQuery<T, Q extends JPQLNextQuery<T, Q>> extends JPQLQuery<T> {
+public interface JPQLNextQuery<T> extends JPQLQuery<T>, ExtendedFetchable<T> {
```

Comment:
I know it sucks, but do you think it's possible to keep around this type variable to stay compatible with the old version? We can remove the type variable in 2.0 if you want. If you say this is necessary though, I'm fine with breaking user code, but AFAICT, it's a cosmetic change.

Reply:
I don't think anyone currently uses the `JPQLNextQuery` class directly, because everywhere we just returned `BlazeJPAQuery` and none of our examples mentioned this API. None of my codebases interacts with  `JPQLNextQuery` directly currently.

Reply:
We can postpone the signature change though, but then I think in 2.0 the impact will be slightly more drastic, because with the new `JPQLQueryFactory` interface people might actually start programming against `JPQLNextQuery<T, ?>` instead of `BlazeJPAQuery<T>` which would then later need to be replaced with `JPQLNextQuery<T>`.

Reply:
I think eventually I would like to separate `-api` and `-impl` as separate modules in 2.0 (also to get rid of the provided scoped dependency). This is kind of a step towards that goal by properly separating API interfaces from implementation.

Reply:
Sounds ok to me. Please just add a note to the changelog that people will probably have to recompile code when using the new version.